client/components/pages/ManageRolesPage.jsx

```diff
@@ -0,0 +1,229 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { Link } from 'react-router';
+import toastr from 'toastr';
+import Divider from 'material-ui/Divider';
+import { Card } from 'material-ui/Card';
+import FlatButton from 'material-ui/FlatButton';
+import { Table, TableBody, TableHeader, TableHeaderColumn,
+  TableRow, TableRowColumn } from 'material-ui/Table';
+import Nav from '../layouts/Nav.jsx';
+import Sidebar from '../layouts/Sidebar.jsx';
+import * as roleActions from '../../actions/roleActions';
+import insertRole from '../../utils/insertRole';
+import TextInput from '../forms/TextInput.jsx';
+import handleError from '../../utils/errorHandler';
+
+class ManageRolesPage extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      roles: [...props.roles],
+      newRole: { title: '' },
+      editRole: { title: '' },
+      roleToEdit: ''
+    };
+    this.editButtonClick = this.editButtonClick.bind(this);
+    this.handleEditChange = this.handleEditChange.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.onEditSubmit = this.onEditSubmit.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  componentWillMount() {
+    this.props.actions.getRoles();
+  }
+
+  componentDidMount = () => {
+    $('.modal').modal();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.state.roles !== nextProps.roles) {
+      this.setState({
+        roles: [...nextProps.roles]
+      });
+    }
+  }
+
+  handleChange(event) {
+    if (!(/[^A-Za-z]/).test(event.target.value)) {
```

Comment:
If you will be using several regex, you can create a regex constant file and import from there.
Otherwise, store these regex in a constant first then use where needed.